mosaic-runtime/src/main/kotlin/com/jakewharton/mosaic/Static.kt

```diff
@@ -0,0 +1,108 @@
+package com.jakewharton.mosaic
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.ComposeNode
+import androidx.compose.runtime.DisallowComposableCalls
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.remember
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+
+/**
+ * Will render each value emitted by [items] as permanent output above the
+ * regular display.
+ */
+@Composable
+fun <T> Static(
+	items: Flow<T>,
+	content: @Composable (T) -> Unit,
+) {
+	class Item(val value: T, var rendered: Boolean)
+
+	// Keep list of items which have not yet been rendered.
+	val pending = remember { mutableStateListOf<Item>() }
+
+	LaunchedEffect(items) {
+		items.collect {
+			pending.add(Item(it, rendered = false))
+		}
+	}
+
+	Static(
+		postRender = {
+			// Remove any items which have been rendered.
+			pending.removeIf { it.rendered }
+		}
+	) {
+		for (item in pending) {
+			Row {
+				// Render item and mark it as having been included in render.
+				content(item.value)
+				item.rendered = true
+			}
+		}
+	}
+}
+
+/**
+ * Renders [content] permanently above the normal canvas. When content has
+ * actually been written to a [TextCanvas], the [postRender] callback will be
+ * invoked to allow clearing of content.
+ *
+ * @param postRender Callback after rendering to a [TextCanvas] is complete.
+ * @param content Content which should be rendered permanently above normal
+ * canvas.
+ */
+@Composable
+internal fun Static(
+	postRender: @DisallowComposableCalls () -> Unit = {},
```

Comment:
```suggestion
	postRender: () -> Unit = {},
```
This should only be needed if the enclosing function was inline.